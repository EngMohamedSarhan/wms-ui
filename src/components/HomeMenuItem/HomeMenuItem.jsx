import { StarBorder } from "@mui/icons-material";
import { Button, Card, Grid, IconButton, Typography } from "@mui/material";
import PropTypes from "prop-types";
import React from "react";
import { useNavigate } from "react-router";

const HomeMenuItem = ({ to, label, description, ...props }) => {
  const navigate = useNavigate();

  const handleNavigate = () => {
    navigate(to);
  };

  return (
    <Grid item {...props}>
      <Card sx={{ p: 2 }}>
        <Typography variant="h4" fontWeight="600">
          {label}
        </Typography>
        <Grid item p={1} />
        <Typography variant="p" fontWeight="400">
          {description}
        </Typography>
        <Grid item p={1} />
        <Grid container alignItems="center" justifyContent="space-between">
          <Grid item>
            <Button variant="contained" onClick={handleNavigate}>
              View and Manage
            </Button>
          </Grid>
          <Grid item>
            <IconButton>
              <StarBorder />
            </IconButton>
          </Grid>
        </Grid>
      </Card>
    </Grid>
  );
};

HomeMenuItem.propTypes = {
  to: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default HomeMenuItem;
