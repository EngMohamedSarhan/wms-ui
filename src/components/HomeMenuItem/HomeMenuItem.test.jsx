/* eslint-disable no-undef */
import { fireEvent, render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import React from "react";
import { Router } from "react-router-dom";
import ShallowRenderer from "react-shallow-renderer";

import HomeMenuItem from "./HomeMenuItem";

describe("HomeMenuItem Component", () =>
  test("HomeMenuItem action", () => {
    const history = createMemoryHistory({ initialEntries: ["/"] });
    const { container } = render(
      <Router location={history.location} navigator={history}>
        <HomeMenuItem to="/lpn-date-check" label="test" description="test" />
      </Router>
    );
    expect(history.location.pathname).toBe("/");
    const button = container.getElementsByTagName("button")[0];
    fireEvent.click(button);
    expect(history.location.pathname).toBe("/lpn-date-check");
  }));

describe("HomeMenuItem Component", () =>
  test("HomeMenuItem snapshot", () => {
    const renderer = new ShallowRenderer();
    const history = createMemoryHistory({ initialEntries: ["/lpn-date-check"] });
    renderer.render(
      <Router location={{ ...history.location, key: "1" }}>
        <HomeMenuItem to="/lpn-date-check" label="test" description="test" />
      </Router>
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  }));
