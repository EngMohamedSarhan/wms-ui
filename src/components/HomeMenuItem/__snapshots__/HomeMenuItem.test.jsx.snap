// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeMenuItem Component HomeMenuItem snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "basename": "/",
      "navigator": undefined,
      "static": false,
    }
  }
>
  <Context.Provider
    value={
      Object {
        "location": Object {
          "hash": "",
          "key": "1",
          "pathname": "/lpn-date-check",
          "search": "",
          "state": null,
        },
        "navigationType": "POP",
      }
    }
  >
    <HomeMenuItem
      description="test"
      label="test"
      to="/lpn-date-check"
    />
  </Context.Provider>
</Context.Provider>
`;

exports[`HomeMenuItem snapshot HomeMenuItem snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "basename": "/",
      "navigator": undefined,
      "static": false,
    }
  }
>
  <Context.Provider
    value={
      Object {
        "location": Object {
          "hash": "",
          "key": "1",
          "pathname": "/lpn-date-check",
          "search": "",
          "state": null,
        },
        "navigationType": "POP",
      }
    }
  >
    <HomeMenuItem
      description="test"
      label="test"
      to="/lpn-date-check"
    />
  </Context.Provider>
</Context.Provider>
`;
